src/JustEat.StatsD/PooledUdpTransport.cs

```diff
@@ -57,20 +42,23 @@ public void Send(string metric)
             var endpoint = _endpointSource.GetEndpoint();
 
             var socket = _socketPool.Pop();
+            socket.SendTo(bytes, endpoint);
```

Comment:
This will leak the socket if `SendTo()` throws, so should be:

```csharp
var socket = _socketPool.Pop();

try
{
    socket.SendTo(bytes, endpoint);
}
catch (Exception)
{
    socket.Dispose();
    throw;
}

_socketPool.Push(socket);
```